pkg/plugin/admissioncontrollers/webhook/mutating_controller.go

```diff
@@ -95,6 +94,48 @@ func (c *admissionController) mutate(ar *v1beta1.AdmissionReview) *v1beta1.Admis
 	}
 }
 
+func updateSchedulerName(patch []patchOperation) []patchOperation {
+	log.Logger.Info("updating scheduler name")
+	return append(patch, patchOperation{
+		Op:    "add",
+		Path:  "/spec/schedulerName",
+		Value: conf.GetSchedulerConf().SchedulerName,
+	})
+}
+
+func updateLabels(pod *v1.Pod, patch []patchOperation) []patchOperation {
+	log.Logger.Info("updating pod labels")
+	existingLabels := pod.Labels
+	result := make(map[string]string)
+	for k,v := range existingLabels {
+		result[k] = v
+	}
+
+	if _, ok := existingLabels[common.SparkLabelAppId]; !ok {
+		if _, ok := existingLabels[common.LabelApplicationId]; !ok {
+			// if app id not exist, generate one
+			generatedId := fmt.Sprintf("autogen_%s_%s", pod.Name, uuid.New().String())
+			log.Logger.Debug("adding application ID",
+				zap.String("generatedID", generatedId))
+			result[common.LabelApplicationId] = generatedId
+		}
+	}
+
+	if _, ok := existingLabels[common.LabelQueueName]; !ok {
+		log.Logger.Debug("adding queue name",
+			zap.String("defaultQueue", "root.default"))
+		result[common.LabelQueueName] = "root.default"
+	}
```

Comment:
The condition here is to make sure when the user explicitly sets queue names for their pods, we don't overwrite that. This needs to work with or without placement rule enabled. When placement rule is enforced, queues set here will be ignored.

Reply:
The only reason I was mentioning this was: should we set it when we create the app object in the scheduler if not set?
Or in other words set a _default_ queue name if it is not set, no matter if it is production code or not.

Reply:
IIUC, you are suggesting to move this logic to our scheduler itself, by setting a default queue if it is not found in the pod's spec. If this is the case, I still prefer to handle this in the admission controller. The reason is this way we are able to mutate pod's spec in a way end users can see, they can see these fields by describing a pod. Otherwise, this info will be hidden by the scheduler. If a pod somehow could not be scheduled, they will find it's hard to diagnosis because of lacking information. Does that make sense to you?

Reply:
OK, I'll link this to the cloudera/yunikorn-core#33 as that would make this part of this change obsolete